### PR TITLE
[None][doc] Create deployment guide for Llama4 Scout FP8 and NVFP4

### PR DESCRIPTION
**Pull request by chenfeiz0326**
## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive quick start guide for deploying and benchmarking Llama4 Scout 17B models with TensorRT-LLM, including setup, configuration, API usage, evaluation, and troubleshooting instructions.
  * Expanded documentation coverage for quantization options (FP8, NVFP4) and performance benchmarking.

* **Chores**
  * Updated pre-commit configuration to exclude Markdown files from the trailing whitespace check.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
+# Deployment Guide for TensorRT-LLM Llama4 Scout 17B FP8 and NVFP4
```
Comment (nv-guomingz) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:1]: Please refer to this comment https://github.com/NVIDIA/TensorRT-LLM/pull/6543/files#r2247835391

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+# Introduction
+
+This deployment guide provides step-by-step instructions for running the Llama-4-Scout-17B-16E-Instruct model using TensorRT-LLM with FP8 and NVFP4 quantization, optimized for NVIDIA GPUs. It covers the complete setup required; from accessing model weights and preparing the software environment to configuring TensorRT-LLM parameters, launching the server, and validating inference output.
+
+The guide is intended for developers and practitioners seeking high-throughput or low-latency inference using NVIDIA’s accelerated stack—starting with the PyTorch container from NGC, then installing TensorRT-LLM for model serving, FlashInfer for optimized CUDA kernels, and ModelOpt to enable FP8 and NVFP4 quantized execution.
+
+# Access & Licensing
+
+To use Llama4 Scout 17B, you must first agree to Meta’s Llama 4 Community License ([https://ai.meta.com/resources/models-and-libraries/llama-downloads/](https://ai.meta.com/resources/models-and-libraries/llama-downloads/)). NVIDIA’s quantized versions (FP8 and FP4) are built on top of the base model and are available for research and commercial use under the same license.
```
Comment (nv-guomingz) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:13]: FP8 and NVFP4?

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+Docker with NVIDIA Container Toolkit installed
+Python3 and python3-pip (Optional, for accuracy evaluation only)
+
+# Models
+
+* FP8 model: [Llama-4-Scout-17B-16E-Instruct-FP8](https://huggingface.co/nvidia/Llama-4-Scout-17B-16E-Instruct-FP8)
+* NVFP4 model: [Llama-4-Scout-17B-16E-Instruct-FP4](https://huggingface.co/nvidia/Llama-4-Scout-17B-16E-Instruct-FP4)
+
+
+Note that NVFP4 is only supported on NVIDIA Blackwell
```
Comment (nv-guomingz) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:29]: Note that NVFP4 is only supported on NVIDIA Blackwell platform.

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+{"id":"cmpl-bc1393d529ce485c961d9ffee5b25d72","object":"text_completion","created":1753843963,"model":"$MODEL","choices":[{"index":0,"text":" New York is a state located in the northeastern United States. It is bordered by","token_ids":null,"logprobs":null,"context_logits":null,"finish_reason":"length","stop_reason":null,"disaggregated_params":null}],"usage":{"prompt_tokens":6,"total_tokens":22,"completion_tokens":16},"prompt_token_ids":null}
+```
+
+## Troubleshooting Tips
+
+* If you encounter CUDA out-of-memory errors, try reducing max\_batch\_size or max\_seq\_len
+* Ensure your model checkpoints are compatible with the expected format
+* For performance issues, check GPU utilization with nvidia-smi while the server is running
+* If the container fails to start, verify that the NVIDIA Container Toolkit is properly installed
+* For connection issues, make sure port 8000 is not being used by another application
```
Comment (hyukn) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:222]:
```suggestion
* If you encounter CUDA out-of-memory errors, try reducing max\_batch\_size or max\_seq\_len.
* Ensure your model checkpoints are compatible with the expected format.
* For performance issues, check GPU utilization with nvidia-smi while the server is running.
* If the container fails to start, verify that the NVIDIA Container Toolkit is properly installed.
* For connection issues, make sure port 8000 is not being used by another application.
```

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+-p 8000:8000 \
+-v ~/.cache:/root/.cache:rw \
+--name tensorrt_llm \
+nvcr.io/nvidia/tensorrt-llm/release:1.0.0rc4 \
+/bin/bash
+```
+
+Note:
+
+* You can mount additional directories and paths using the \-v \<local\_path\>:\<path\> flag if needed, such as mounting the downloaded weight paths.
```
Comment (hyukn) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:50]:
```suggestion
* You can mount additional directories and paths using the `-v <local_path>:<path>` flag if needed, such as mounting the downloaded weight paths.
```

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+* The command mounts your user .cache directory to save the downloaded model checkpoints which are saved to \~/.cache/huggingface/hub/ by default. This prevents having to redownload the weights each time you rerun the container. If the \~/.cache directory doesn’t exist please create it using  mkdir \~/.cache
```
Comment (hyukn) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:51]: I suggest using inline code `` to wrap this kind of directory or command that is inline. The same below.

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+```shell
+MODEL_PATH=nvidia/Llama-4-Scout-17B-16E-Instruct-FP4
+
+lm_eval --model local-completions  --tasks gsm8k --batch_size 256 --gen_kwargs temperature=0.0 --num_fewshot 5 --model_args model=${MODEL_PATH},base_url=http://localhost:8000/v1/completions,num_concurrent=32,max_retries=20,tokenized_requests=False --log_samples --output_path trtllm.fp4.gsm8k
+```
+
+Sample result in Blackwell
+
+```shell
+|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
```
Comment (hyukn) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:264]: Considering that both commands and accuracy between FP8 and FP4 are close, will it be more compressed if we only keep one of the command descriptions?

File: examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+
+&emsp;&emsp;**Default**: CUTLASS
+
+#### `attention_backend`
+
+&emsp;**Description**: The backend to use for attention calculations.
+
+&emsp;**Default**: TRTLLM
+
+See the [https://github.com/nvidia/TensorRT-LLM/blob/main/tensorrt\_llm/llmapi/llm\_args.py\#L1894](https://github.com/nvidia/TensorRT-LLM/blob/main/tensorrt_llm/llmapi/llm_args.py#L1894) TorchLlmArgs class for the full list of options which can be used in the `extra_llm_api_options`.
```
Comment (nv-guomingz) [inline examples/models/core/llama4/Deployment Guide for TRT-LLM + Llama4 Scout.md:180]: See the [TorchLlmArgs class](https://nvidia.github.io/TensorRT-LLM/llm-api/reference.html#tensorrt_llm.llmapi.TorchLlmArgs) for the full list of options which can be used in the extra\_llm\_api\_options`.`

**Review by litaotju — APPROVED** [review]

**Comment by litaotju** [comment]
Bypassing and merge now. The doc change won't affect CI.
